### DISCUSSION TITLE
Recoil speed

### DISCUSSION THREAD

**Discussion by Vojlok** [rejetto/hfs#837] [Q&A]
Hello, thanks for your great HFS program. 
Tell me how to increase the file upload speed? The server speed is about one gigabit, and files are transferred at a low speed. 
My tariff is 300 megabits, but it ends up being about 20.
I am writing through an online translator, sorry for the language.

**Comment by rejetto** [comment]
300 megabits = 37 megabytes.
what is the 20, megabits or megabytes?

**Reply by Vojlok** [reply]
20 megabits of course

**Comment by Vojlok** [comment]
This is observed not only from my provider, but from others too.
[Here are my files](http://192.162.247.85/)

**Comment by rejetto** [comment]
where did you read that 20 megabits?
you should measure your real speed using a website like fast.com, but you must do it in a moment where you have almost no other internect activity.
i also tried to access your HFS and download a big file, but i got a wide range of speeds, from 1MB/s to 12MB/s, so it's hard to tell anything.

**Reply by Vojlok** [reply]
[https://www.speedtest.net/result/17156991106)](https://www.speedtest.net/result/17156991106)

**Reply by rejetto** [reply]
i made a test right now downloading 8 megabytes /s from you, which is not far from what you measured.
also, i'm in Italy, so you must expect getting a speed that's lower from the speedtest.

**Reply by Vojlok** [reply]
This is the problem, files download at low speed.

**Reply by rejetto** [reply]
8 megabytes is 64 megabits. You measured 89 but with a closer computer, so the lower value with me is expected.
I don't see a problem with HFS.

**Reply by Vojlok** [reply]
Ok, so be it.
Are there any changeable speed output settings?

**Reply by rejetto** [reply]
there are options to restrict the speed, not to increase it

**Comment by TEA-Time1** [comment]
The speed difference between download and upload that you are seeing is very typical for a cable internet connection. Upstream is just not very much with cable, and that's what is used when they download from you. Technically you're uploading it to them.

**Reply by rejetto** [reply]
ah, yes, i forgot to mention that even if you initially siad you have a gigabit with 300mbps, it turns out that your 300 is only for "getting", not "giving", so 300 is NOT the important number here.
When you give (people downloading from you), you only have 89 (as you can see in your link). 89 is your max speed (at the time of the test, at least).

**Reply by PH5HP** [reply]
Hi,

Yres, that's true. I have 400 Mbit download and only about 40Mbit upload... So my theoretical serverspeed of HFS is 40Mbit :-) ... I am also on cable internet here in the Netherlands... The 40Mbit is enough ...

Kind regards,
Henk - fileserver.ph5hp.nl

**Reply by Vojlok** [reply]
> ah, yes, i forgot to mention that even if you initially siad you have a gigabit with 300mbps, it turns out that your 300 is only for "getting", not "giving", so 300 is NOT the important number here. When you give (people downloading from you), you only have 89 (as you can see in your link). 89 is your max speed (at the time of the test, at least).

Server speed is 1 gigabit roundtrip.
My speed is 300 megabits round trip.
==========
It seems that even I should be given at least 100 megabits, but in the end the maximum is 20 megabits per client.
I can't be the only one facing this problem. All updates after that did not fix the problem.
===========
Perhaps you know some tweaks for the server so that it would give up its resource (1 gigabit) and not 20 megabits?

**Reply by rejetto** [reply]
i just tried to download from your hfs, again, and i've got 21MB/s which roughly translates to 168 megabits.
your speed is very strange, anyway, because it occasionally goes much faster, like double that, but it only lasts a few seconds, and then it drops.
if you think it's hfs you should try another software, and see if you get faster speed. If you don't, then the problem is not hfs.
